### ISSUE TITLE
The success rate is very low on real robot even though the open loop testing looks good

### ISSUE THREAD

**Issue by zmingen** [NVIDIA/Isaac-GR00T#414]
### 🐛 Describe the bug

Hi, thanks for open sourcing this excellent work. I am trying to fine tune a model for the pick and place task with unitree G1 robot. We captured about 600 episodes and fine tuned the model, the MSE loss on evaluation dataset looks good.

<img width="2829" height="13797" alt="Image" src="https://github.com/user-attachments/assets/8b37d720-e328-4e91-bb88-4699131a16f6" />

Below is the fine tuning script:

`
dataset_list=(
    "datasets/g1-pick-place-1014-2-cameras-train/"
    "datasets/g1-pick-place-1016-2-cameras-train"
    "datasets/g1-pick-place-1017-2-cameras-train"
    "datasets/g1-pick-place-1020-2-cameras-train"
    "datasets/g1-pick-place-1021-2-cameras-train"
    “datasets/g1-pick-place-1022-2-cameras-train”
)
 
CUDA_VISIBLE_DEVICES="2,3,4,5,6,7" python scripts/gr00t_finetune.py \
    --dataset-path ${dataset_list[@]} \
    --num-gpus 6 --batch-size 8  --output-dir ~/checkpoints/g1-pick-place-1022-2-cameras-train/  \
    --data-config unitree_g1_inspire --max-steps 100000 \
    --video_backend torchvision_av \
    --dataloader_num_workers 6 \
    --save_steps 4000 \
    --report_to tensorboard \
    --tune_visual
`

The unitree_g1_inspire defined as below:
`
class UnitreeG1InspireDataConfig(BaseDataConfig):
    #video_keys = ["video.cam_left_high"]
    #video_keys = ["video.cam_left_high", "video.cam_left_wrist", "video.cam_right_wrist"]
    video_keys = ["video.cam_left_high", "video.cam_left_wrist"]
    #video_keys = ["video.cam_left_high"]
    #state_keys = ["state.left_arm", "state.right_arm", "state.left_hand", "state.right_hand"]
    #action_keys = ["action.left_arm", "action.right_arm", "action.left_hand", "action.right_hand"]
    state_keys = ["state.left_arm", "state.left_hand"]
    action_keys = ["action.left_arm", "action.left_hand"]
    language_keys = ["annotation.human.task_description"]
    observation_indices = [0]
    action_indices = list(range(16))

    def transform(self) -> ModalityTransform:
        transforms = [
            # video transforms
            VideoToTensor(apply_to=self.video_keys),
            VideoCrop(apply_to=self.video_keys, scale=0.95),
            VideoResize(apply_to=self.video_keys, height=224, width=224, interpolation="linear"),
            VideoColorJitter(
                apply_to=self.video_keys,
                brightness=0.3,
                contrast=0.4,
                saturation=0.5,
                hue=0.08,
            ),
            VideoToNumpy(apply_to=self.video_keys),
            # state transforms
            StateActionToTensor(apply_to=self.state_keys),
            StateActionTransform(
                apply_to=self.state_keys,
                normalization_modes={key: "min_max" for key in self.state_keys},
            ),
            # action transforms
            StateActionToTensor(apply_to=self.action_keys),
            StateActionTransform(
                apply_to=self.action_keys,
                normalization_modes={key: "min_max" for key in self.action_keys},
            ),
            # concat transforms
            ConcatTransform(
                video_concat_order=self.video_keys,
                state_concat_order=self.state_keys,
                action_concat_order=self.action_keys,
            ),
            # model-specific transform
            GR00TTransform(
                state_horizon=len(self.observation_indices),
                action_horizon=len(self.action_indices),
                max_state_dim=64,
                max_action_dim=32,
            ),
        ]
        return ComposedModalityTransform(transforms=transforms)
`

### Versions

Some teleoperation episode:

https://github.com/user-attachments/assets/2ef78b09-3657-43e2-b6a3-afe2c0f9aa78

**Comment by HomerIsAFool** [comment]
Hi @zmingen , Could you pls show the inference video on real G1?

**Comment by Young-YangYoung** [comment]
Could you tell me how you collected the data? Do you have any reference code I could look at?
